news/2022-08-25-changes-to-osu-taiko-sr-and-pp.md

```diff
@@ -0,0 +1,143 @@
+
+In addition to the above, specific nerfs for specific patterns were implemented using a dedicated "cheese detector", which was something that the osu!taiko pp committee wanted to stay away from. As such, a policy for the all the new reworks under our wing was formed:
+
+**Never nerf a specific playstyle or pattern solely for a lack of particular reason.**
+
+#### The new stamina system
+
+In this section, the keybinds relating to the two colours found within osu!taiko are named k<sup>1</sup>, k<sup>2</sup> and d<sup>1</sup>, d<sup>2</sup>.
+
+Instead of assuming a full-alternate playstyle, we assume that **each note of the same colour is always alternated**. For example, for the pattern kddkddk, we assume the kats will be hit by **k<sup>1</sup>** d<sup>1</sup> d<sup>2</sup> **k<sup>2</sup>** d<sup>1</sup> d<sup>2</sup> **k<sup>1</sup>**, and the dons will be hit by k<sup>1</sup> **d<sup>1</sup> d<sup>2</sup>** k<sup>2</sup> **d<sup>1</sup> d<sup>2</sup>** k<sup>2</sup>.
+
+The reason for this is to assume the best-case scenario for repeated keystrokes regardless of playstyle. In other words, we aim to approximate the theoretical minimum stamina required to play a given pattern with any playstyle, with any theoretically rollable pattern being rolled. We can't (feasibly), nor is it desirable to, detect playstyles, so we assume the case that requires the least physical repetition, which is to switch keys each time a colour is to be hit, and as such all other existing playstyles will be weaker in comparison to the one we chose.
+
+These changes also help with cases where specific playstyles or patterns were nerfed in the previous rework, as we no longer detect any patterns, stopping at calculating the pure stamina requirement for each map.
+
+### Readdressing colour
+
+The old colour system detected the frequency of colour changes and had penalties for specific repeated patterns. This meant that only notes of a different colour than the previous one would receive a non-zero strain, which contributed to issues of complex patterns containing large groups of single-coloured notes not being considered properly in the strain system. Similar to stamina, a penalty was applied to repeated occurrences of single-coloured patterns. While only the 5 most recent occurrences were considered in this penalty, if any two or more of those patterns were repeated, a harsh penalty was applied.
+
+#### The new colour system
+
+In the new system, we define colour difficulty as the amount of information contained within a given pattern (which can be thought as of a sequence of notes). For example, we can take a look at the following colour patterns and roughly estimate the amount of information required to describe them:
+
+![Colour pattern table showcasing colour sequence compression](/wiki/shared/news/2022-08-25-changes-to-osu-taiko-star-rating-and-performance-points/colour-patterns.jpg)
```

Comment:
i'm not sure why repetitions are represented as superscripts, why not multiplication sign and number?

Reply:
That was done mainly for the looks, as the table in the doc looked a bit messy. It also somewhat better portray the "amount of information" bit, as multiplication sign + number makes repetition number looks a little long (i.e. `DDD` looks simpler than `D * 3` or `D x 3`).

As a side note we also considered just number followed by the note (i.e. `5D` etc.), but that was not done due to this font having the large dot symbol not aligned with the number.

I'll pass this to the committee, if it's agreed that multiplication sign + number is more straightforward I'll commit the changes

Reply:
> i.e. DDD looks simpler than D * 3 or D x 3

i'd like to disagree since the left column of the table is supposed to show the "uncompressed" version, i.e. DDD, while Dx3 would be "compressed"

Reply:
The idea is to make the "compressed" version be as short as possible. That said I can see how a more straightforward notation without the need a legend for description could be a better approach, as the purpose of the table is to give a rough "amount of information" view on the sequences, instead of actually trying to compress the sequences to a minimal amount of characters.

LTCA seems to prefer the shorter version with legend, though that is expected as he was involved in the creation. I'll ask around a bit more and see which one people prefers

Reply:
It seems like the shorter notation is preferred generally. Mangomizer suggested using notation like 4D, 4K, which makes algebraic sense, and might remove the need for the legend. What do you think about the looks of the attached table?
![colour-table](https://user-images.githubusercontent.com/5846322/190328903-adf3b418-6936-42ad-88f0-a30240b3b4b7.png)

Reply:
looks ok but please do include legend for D and K if you're also doing C, to make things more obvious for people skimming the article

Reply:
also would be better to clarify that you're *flipping* the colours, that a C represents a flipped copy of the previous pattern

Reply:
Sounds good, attached again for a final confirmation before commit:
![colour-table](https://user-images.githubusercontent.com/5846322/190366152-74c8346a-0502-4760-84da-2f3f8ad27b27.png)

Reply:
looks good but just "don (red) note" is fine

Reply:
Applied in adcb1fcdb1636954e34d876f03609fc83a098f66